### PR TITLE
Add support for Symfony 6

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

Part of: https://github.com/sonata-project/dev-kit/issues/1675
Dev-kit PR: https://github.com/sonata-project/dev-kit/pull/1684

## Changelog

```markdown
### Added
- Added explicit dependencies with Doctrine and Symfony.
- Added support for Symfony 6
### Removed
- Removed support for Symfony 5.2
```

### CODE REVIEW

**Review by phansys — COMMENTED** [review]
About the "symfony/http-foundation" dependency, I think this should be optional, since this bundle can work without its controllers.

**Comment by jordisala1991** [comment]
> About the "symfony/http-foundation" dependency, I think this should be optional, since this bundle can work without its controllers.

Yep, I added it to require dev for that matter.

**Review by phansys — APPROVED** [review]

File: src/Collection/AuditedCollection.php [hunk 1]
```diff
@@ -105,6 +105,9 @@ public function __construct(AuditReader $auditReader, $class, ClassMetadataInfo
         $this->entities = new ArrayCollection();
     }
 
+    /**
+     * @return bool
```
Comment (VincentLanglet) [inline src/Collection/AuditedCollection.php:109]: Why can't we rely on interface phpdoc ?

**Review by VincentLanglet — APPROVED** [review]

File: src/Collection/AuditedCollection.php (hunk 1 above)
Comment (jordisala1991) [reply src/Collection/AuditedCollection.php:109]: If you don't add it here, you get deprecation from symfony. This way we silence it (and we know that at some point we will need to add it to the method directly without phpdoc.

**Pull request merged**